operator/pkg/controller/karmada/controller.go

```diff
@@ -160,6 +160,7 @@ func (ctrl *Controller) SetupWithManager(mgr controllerruntime.Manager) error {
 			builder.WithPredicates(predicate.Funcs{
 				UpdateFunc: ctrl.onKarmadaUpdate,
 			})).
+		WithEventFilter(skipIfSuspended()).
```

Comment:
Put the event filter into `For`?  `builder.WithPredicates(skipIfSuspended())`

No big deal though. 
Note that the EventFileter will be the global filter that works for `Owns` and `Watches`, in addition to `For`. We might don't need the `Owns` and `Watches` now, but once they are involved in the future, the event filter probably not work, because the object type passing in the function might not be `Karmada`.

For the operator controller, we might don't need to use `Owns` and `Watches`, but once they are used, the event  filter w